integration-tests/test_e2e.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import conftest
+from constants import MACHINE_ID_FILE
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_insights_client_version_in_inventory(insights_client, external_inventory):
+    """
+    Verify running insights-client creates a new host on Inventory and the host has
+    the insights client and egg version information available
+
+        test_steps:
+            1. Register the system to Insights
+            4. Retrieve the system profile from Inventory
+        expected_results:
+            1. Successful registration to insights server
+            2. The system is successfully retrieved from Inventory
+            3. system data retrieved from Inventory has client and egg version info
+    """
+    insights_client.run("--register")
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id = f.read()
+    # fetching host details to get host ID
+    response = external_inventory.get(path=f"hosts?insights_id={machine_id}")
+    host_id = response.json()["results"][0]["id"]
+    # Using host ID to get system_profile
+    response = external_inventory.get(path=f"hosts/{host_id}/system_profile")
```

Comment:
hm this should rather be added to pytest-client-tools

Reply:
yes, makes sense. I will remove it from here

Reply:
added as new method `Inventory.this_system_profile()`